### PR TITLE
feature: OSIS-83 add gradle closeAndRelease to nexus step in release …

### PR DESCRIPTION
**Pull request by XinLiScality**
…workflow

extracted `gradle-build-and-upload` step to a dedicated file, and use it in both test workflow and release workflow but diff them by has or not `release` flag.

### CODE REVIEW

File: upload-artifact.gradle [hunk 1]
```diff
@@ -1,5 +1,35 @@
 apply plugin: 'com.bmuschko.nexus'
 apply plugin: 'signing'
+
+modifyPom {
```
Comment (WillToozsScality) [inline upload-artifact.gradle:4]: Can you explain to me quickly what this part is for?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -35,3 +35,10 @@ jobs:
           tag_name: ${{ github.event.inputs.tag }}
           generate_release_notes: true
           target_commitish: ${{ github.sha }}
+
+  gradle_build_and_upload:
```
Comment (williamlardier) [inline .github/workflows/release.yml:39]: Nit, if you think that makes sense, to keep the same formatting:
```suggestion
  gradle-release:
```

File: .github/workflows/gradle-build-and-upload.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      release:
+        required: false
+        type: boolean
+        default: false
+
+jobs:
+  gradle_build_and_upload:
+    name: Gradle build and upload
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
```
Comment (williamlardier) [inline .github/workflows/gradle-build-and-upload.yml:15]: Just curious why we enforce a timeout here? Are we expecting flakiness?

**Review by williamlardier — APPROVED** [review]

File: upload-artifact.gradle (hunk 1 above)
Comment (XinLiScality) [reply upload-artifact.gradle:4]: this modifyPom part is to build a POM XML file that meets the maven standard of released packages, a POM XML file should contain the information of project and configuration information for the maven to build the project

File: .github/workflows/gradle-build-and-upload.yml (hunk 3 above)
Comment (XinLiScality) [reply .github/workflows/gradle-build-and-upload.yml:15]: `gradle build` takes some time to finish, it solves all the dependencies and runs all the test.

**Review by WillToozsScality — APPROVED** [review]

**Pull request merged**